apps/server/src/adapters/WebsocketAdapter.ts

```diff
@@ -40,70 +41,116 @@ export class SocketServer implements IAdapter {
+      ws.send(
+        JSON.stringify({
+          type: 'ontime',
+          payload: eventStore.poll(),
         }),
       );
 
       ws.on('error', console.error);
 
       ws.on('close', () => {
-        this.clientIds.delete(clientId);
-        logger.info(LogOrigin.Client, `${this.clientIds.size} Connections with disconnected: ${clientId}`);
+        this.clients.delete(clientId);
+        logger.info(LogOrigin.Client, `${this.clients.size} Connections with disconnected: ${clientId}`);
+        this.sendClientList();
       });
 
       ws.on('message', (data) => {
         try {
-          // @ts-expect-error -- ??
+          //@ts-expect-error ???
           const message = JSON.parse(data);
           const { type, payload } = message;
 
           if (type === 'get-client-name') {
             ws.send(
               JSON.stringify({
                 type: 'client-name',
-                payload: clientId,
+                payload: this.clients.get(clientId).name,
               }),
             );
             return;
           }
 
           if (type === 'set-client-name') {
             if (payload) {
-              const previousName = clientId;
-              clientId = payload;
-              this.clientIds.delete(previousName);
-              this.clientIds.add(clientId);
-              logger.info(LogOrigin.Client, `Client ${previousName} renamed to ${clientId}`);
+              const previousData = this.clients.get(clientId);
+              logger.info(LogOrigin.Client, `Client ${previousData.identify} renamed to ${payload}`);
+              previousData.name = payload;
+              this.clients.set(clientId, previousData);
             }
             ws.send(
               JSON.stringify({
                 type: 'client-name',
-                payload: clientId,
+                payload,
               }),
             );
+            this.sendClientList();
+            return;
+          }
+
+          if (type === 'set-client-identify') {
+            if (payload) {
+              const targetClient = this.clients.get(payload.target);
+              this.clients.set(payload.target, { ...targetClient, identify: payload.state });
+              this.sendClientList();
+            }
+            return;
+          }
+
+          if (type === 'set-client-redirect') {
+            if (payload) {
+              const targetClient = this.clients.get(payload.target);
+              const redirect = payload.path;
+              this.clients.set(payload.target, { ...targetClient, redirect });
+            }
+            this.sendClientList();
+            return;
+          }
+
+          if (type === 'set-client-rename') {
+            if (payload) {
+              const targetClient = this.clients.get(payload.target);
+              const rename = payload.name;
+              this.clients.set(payload.target, { ...targetClient, rename });
+            }
+            this.sendClientList();
             return;
           }
```

Comment:
suggestions: 
- Would it be easier to maintain this feature if we extracted the functions here?
- Can we have a timeout for the client identify to stop someone forgetting this on and a screen staying blinking? ie: after 10 seconds an identify command resets itself.

Reply:
i'm thinking about how long to set the timeout for...
it would be anoying to set the idetify walk over to the screens and the it has reset befor you get there
maybe atleast 30sec or a whole minute

Reply:
I am not sure, I think 1 minute sounds reasonable to me.

I think if the venue is so large that you need a long time to walk around, we likely want either another solution or some help